crates/bevy_scene/src/scene_spawner.rs

```diff
@@ -268,11 +268,25 @@ impl SceneSpawner {
         for (instance_id, parent) in scenes_with_parent {
             if let Some(instance) = self.spawned_instances.get(&instance_id) {
                 for entity in instance.entity_map.values() {
+                    // Add the `Parent` component to the scene root
                     if let Some(mut entity_mut) = world.get_entity_mut(entity) {
+                        // This will filter only the scene root entity, as all other from the
+                        // scene have a parent
                         if !entity_mut.contains::<Parent>() {
                             entity_mut.insert(Parent(parent));
                         }
                     }
+                    // Add the scene root to the `Children` component on the parent
+                    if let Some(mut parent_entity) = world.get_entity_mut(parent) {
```

Comment:
This feels like quite the hack, both because we're duplicating parenting logic in scenes and because of the children copy. I want to fix this bug, but maybe we need a slightly more holistic solution. Hard-coding logic here seems like a reasonable short term fix (with a TODO to improve this later), but reallocating Children for each child in a scene is hard to accept. One way to implement this without directly exposing internals is to add a new `AddChild` command to `bevy_transform`, which takes advantage of crate-only access to the Children internals. Then you could just construct the Command for each entity and immediately apply it (no need for command buffers / allocations).

Reply:
(AddChild would transactionally set the proper values for the Parent and Children components)

Reply:
added `AddChild` command. It's the only command with pub fields to be able to create and execute it directly